lib/Segments/DynamicSegments/DynamicSegmentsListingRepository.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace MailPoet\Segments\DynamicSegments;
+
+use MailPoet\Entities\SegmentEntity;
+use MailPoet\Segments\SegmentListingRepository;
+
+class DynamicSegmentsListingRepository extends SegmentListingRepository {
+  protected $types = [SegmentEntity::TYPE_DYNAMIC];
```

Comment:
I think that instead of inheritance we could pass this as an argument or via a constructor or a setter.